Added support for oauth2 authentication type.

taylorsteffanj opened the pull request.

Commit:
Added support for oauth2 authentication type.

elljoh commented on ibmpairs/paw.py:
```diff
@@ -498,15 +513,20 @@ def __init__(
             self.publish2GUI    = False
         # set GUI password (default to PAIRS API password)
-        self.guiPassword        = guiPassword if guiPassword is not None else self.auth[1]
+        if type(auth) is authentication.OAuth2:
```
`auth` may be `None` here when `self.auth` is set on Line 493

cmalbrec replied:
we fix by `auth` to `self.auth` in line 516

cmalbrec commented:
We will resolve https://travis-ci.com/github/IBM/ibmpairs/jobs/482758997 by sunset of support for Python 2 which reached its end-of-life more than a year ago (https://www.python.org/doc/sunset-python-2/).

cmalbrec committed:
discard support for Python 2

... removing Travis tests

DCO 1.1 Signed-off-by: Conrad M Albrecht <cmalbrec@us.ibm.com>

cmalbrec committed:
Bug fix: instantiation of authentication object

DCO 1.1 Signed-off-by: Conrad M Albrecht <cmalbrec@us.ibm.com>

cmalbrec committed:
update copyright and contributors information

plus cleaning of whitespace

DCO 1.1 Signed-off-by: Conrad M Albrecht <cmalbrec@us.ibm.com>

cmalbrec committed:
Bug fix: keep query result publication to GUI

separate from from PAIRS backend authentication

DCO 1.1 Signed-off-by: Conrad M Albrecht <cmalbrec@us.ibm.com>

cmalbrec commented on tests/test_paw.py:
```diff
@@ -123,24 +127,33 @@
-            PAIRS_SERVER, PAIRS_BASE_URI, PAIRS_USER, PAIRS_PASSWORD_FILE_NAME,
-        )
-    )
+if AUTH_TYPE.lower() in ['api-key', 'apikey', 'api key']:
+    OAUTH = authentication.OAuth2(host         = PAIRS_SERVER,
+                                  username     = PAIRS_USER,
+                                  api_key_file = PAIRS_PASSWORD_FILE_NAME
+                                 )
+    PAIRS_JWT_TOKEN = OAUTH.jwt_token
+    PAIRS_CREDENTIALS       = (PAIRS_USER, PAIRS_PASSWORD,)
```
it might be useful to add some logging to track testing

cmalbrec committed:
update copyright info II

DCO 1.1 Signed-off-by: Conrad M Albrecht <cmalbrec@us.ibm.com>

cmalbrec reviewed (approved):
modification to existing code `paw.py` and its tests `test_paw.py` reviewed for backward compatibility

Commit:
Added Bearer to Publish

Commit:
Added Bearer to Publish
